OSXDemo/OCTConversationViewController.m

```diff
@@ -0,0 +1,262 @@
+
+- (void)controllerWillChangeContent:(RBQFetchedResultsController *)controller
+{
+    NSTableView *tableView = (self.chatResultsController == controller) ? self.chatsTableView : self.conversationTableView;
+    [tableView beginUpdates];
+}
+
+- (void) controller:(RBQFetchedResultsController *)controller
+    didChangeObject:(RBQSafeRealmObject *)anObject
+        atIndexPath:(NSIndexPath *)indexPath
+      forChangeType:(RBQFetchedResultsChangeType)type
+       newIndexPath:(NSIndexPath *)newIndexPath
+{
+    NSTableView *tableView = (self.chatResultsController == controller) ? self.chatsTableView : self.conversationTableView;
+
+    NSIndexSet *newSet = [[NSIndexSet alloc] initWithIndex:newIndexPath.row];
+    NSIndexSet *oldSet = [[NSIndexSet alloc] initWithIndex:indexPath.row];
+    NSIndexSet *columnSet = [[NSIndexSet alloc] initWithIndex:0];
+
+
+    switch (type) {
+        case RBQFetchedResultsChangeInsert:
+            [tableView insertRowsAtIndexes:newSet withAnimation:NSTableViewAnimationSlideRight];
+            break;
+        case RBQFetchedResultsChangeDelete:
+            if (tableView == self.chatsViewController) {
+                [tableView removeRowsAtIndexes:oldSet withAnimation:NSTableViewAnimationSlideLeft];
+                [self selectFirstChat];
+            }
+            break;
+        case RBQFetchedResultsChangeUpdate:
+            [tableView reloadDataForRowIndexes:oldSet columnIndexes:columnSet];
+            break;
+        case RBQFetchedResultsChangeMove:
+            [tableView removeRowsAtIndexes:oldSet withAnimation:NSTableViewAnimationSlideLeft];
+            [tableView insertRowsAtIndexes:newSet withAnimation:NSTableViewAnimationSlideRight];
+            break;
+    }
+}
+
+- (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
+{
+
+    NSTableView *tableView = (self.chatResultsController == controller) ? self.chatsTableView : self.conversationTableView;
+
+    @try {
+        [tableView endUpdates];
+    }
+    @catch (NSException *ex) {
+        [controller reset];
+        [tableView reloadData];
+    }
+}
+
+#pragma mark - Private
+
+- (void)updateConversationControllerForChat:(OCTChat *)chat
+{
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"chat.uniqueIdentifier == %@", chat.uniqueIdentifier];
+    RBQFetchRequest *fetchRequest = [self.manager.objects fetchRequestForType:OCTFetchRequestTypeMessageAbstract withPredicate:predicate];
+    [self.conversationResultsController updateFetchRequest:fetchRequest
+                                        sectionNameKeyPath:nil
+                                            andPeformFetch:YES];
+}
+
+- (void)selectFirstChat
+{
+    NSIndexPath *path = [NSIndexPath indexPathForRow:0 inSection:0];
+    OCTChat *chat = [self.chatResultsController objectAtIndexPath:path];
+
+    if (chat) {
+        NSIndexSet *indexSet = [[NSIndexSet alloc] initWithIndex:0];
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
I see. Can you then add famous comment (hopefully this will be fixed one day).

```objc
// workaround for deadlock in objcTox
// https://github.com/Antidote-for-Tox/objcTox/issues/51
```